amundsen_application/api/utils/metadata_utils.py

```diff
@@ -228,7 +228,7 @@ def _map_user_object_to_schema(u: Dict) -> Dict:
 
 def _get_partition_data(watermarks: Dict) -> Dict:
     if watermarks:
-        high_watermark = next(filter(lambda x: x['watermark_type'] == 'high_watermark', watermarks))
+        high_watermark = next(filter(lambda x: x['watermark_type'].lower() == 'high_watermark', watermarks))
```

Comment:
I see why this fix works, but if so inclined we could use `amundsen_common` for declaring these sort of constants that all services have to recognize. Could you file an issue and use this bug as an example of the problems we run into hard-coding the values across different repos? Thanks!